### ISSUE TITLE
What if I press ENTER for email while adduser

### ISSUE THREAD

**Issue by zhanzhenzhen** [npm/npm#6126]
I don't know if ENTER means "no change". So I just input the same email address while `adduser` in a new machine. Why not make the hint more informative?

**Comment by smikes** [comment]
I believe this has been addressed sometime in the last few months.  When I use `npm login` with `npm@latest`, I am prompted like this:

```
Username: (smikes) 
Password: (or leave unchanged) 
Email: (this IS public) (smikes@cubane.com) 
```

which shows my previously-configured email address as the default value for `email`.

@zhanzhenzhen is this what you had in mind by making the hint more informative, or do you think it could still be better?

**Comment by zhanzhenzhen** [comment]
I think it's not your first login on this machine. I just mean first login on a machine.

Email is different than username and password. I think better the email field has this `or leave unchanged` hint, on first login on a machine.

But when I reviewed the `adduser` help, I found:
"When authenticating on a new machine, the username, password and email address must all match with your existing record."
I don't know why email address must also match. username and password is enough. Email field is useless and confusing here.

**Comment by smikes** [comment]
Oh, yes, that's a different problem.  At present the username, password, and email must all match when logging in from a new machine.  That may be going to be fixed in registry-2.0, but I haven't heard about that.

**Comment by othiym23** [comment]
Some of the registry backends require the email address as part of authentication (this is a holdover from the days when all of the registries were implemented as CouchDB applications). It seems fairly strange to newcomers that a third piece of information other than username and password is required upon login (rather than account creation), but even when we split apart user creation and user authentication (by splitting `npm adduser` and `npm login`), we still aren’t going to have the information at login time that allows us to decide whether the back-end needs the email address, or is using the new authentication strategy used by the primary npm registry, which consumes usernames and passwords and returns bearer tokens.

Implementing automatic detection of which auth strategy is being used is surprisingly tricky, given the number of different registries out there, and as such isn’t something that we’re going to have the resources to do in the near term. As such, I’m going to close this issue, although I would like to eventually have a better solution for this in the CLI. Thanks for your time!
